hq/test/aws/ec2/EC2Test.scala

```diff
@@ -140,6 +141,22 @@ class EC2Test extends FreeSpec with Matchers with Checkers with PropertyChecks w
         sortedResult should be(sortedResult.sortWith { case ((_, s1), (_, s2)) => s1.size > s2.size })
       }
     }
+
+    "add cloudformation stackId & stackName" in {
+      val sgs1 = new SecurityGroup().withGroupId("id-1").withTags(new Tag().withKey("aws:cloudformation:stack-id").withValue("stack-id-1"), new Tag().withKey("aws:cloudformation:stack-name").withValue("stack-name-1"))
+      val sgs2 = new SecurityGroup().withGroupId("id-2").withTags(new Tag().withKey("aws:cloudformation:stack-id").withValue("stack-id-2"), new Tag().withKey("aws:cloudformation:stack-name").withValue("stack-name-2"))
+      val sgs3 = new SecurityGroup().withGroupId("id-3")
+      val result = new DescribeSecurityGroupsResult().withSecurityGroups(sgs1, sgs2, sgs3)
+      val sgsOpenPort1 = SGOpenPortsDetail("warning", "eu-west-1", "name-1", "id-1", "vpc-1", "tcp", "8080", "Red", false )
+      val sgsOpenPort2 = SGOpenPortsDetail("warning", "eu-west-2", "name-2", "id-2", "vpc-2", "tcp", "8080", "Red", false )
+      val sgsOpenPort3 = SGOpenPortsDetail("warning", "eu-west-1", "name-4", "id-4", "vpc-4", "tcp", "8080", "Red", false )
+      val account = AwsAccount("id-1", "security", "security-role")
+      EC2.updateSgsWithTags(account, List(sgsOpenPort1, sgsOpenPort2, sgsOpenPort3))((_, _) => Attempt.Right(result) ).value().toSet shouldBe
+        Set(
+        sgsOpenPort1.copy(stackId = Some("stack-id-1"), stackName = Some("stack-name-1")),
+        sgsOpenPort2.copy(stackId = Some("stack-id-2"), stackName = Some("stack-name-2")),
+        sgsOpenPort3)
+    }
```

Comment:
Will it be okay if there is a Security Group with a stackId, but no stackName?

Tried this:
```
    "add cloudformation stackId & stackName" in {
      val sgs1 = new SecurityGroup().withGroupId("id-1").withTags(new Tag().withKey("aws:cloudformation:stack-id").withValue("stack-id-1"), new Tag().withKey("aws:cloudformation:stack-name").withValue("stack-name-1"))
      val sgs2 = new SecurityGroup().withGroupId("id-2").withTags(new Tag().withKey("aws:cloudformation:stack-id").withValue("stack-id-2"), new Tag().withKey("aws:cloudformation:stack-name").withValue("stack-name-2"))
      val sgs3 = new SecurityGroup().withGroupId("id-3").withTags(new Tag().withKey("aws:cloudformation:stack-id").withValue("stack-id-3"))
      val sgs4 = new SecurityGroup().withGroupId("id-4")
      val result = new DescribeSecurityGroupsResult().withSecurityGroups(sgs1, sgs2, sgs3, sgs4)
      val sgsOpenPort1 = SGOpenPortsDetail("warning", "eu-west-1", "name-1", "id-1", "vpc-1", "tcp", "8080", "Red", false )
      val sgsOpenPort2 = SGOpenPortsDetail("warning", "eu-west-2", "name-2", "id-2", "vpc-2", "tcp", "8080", "Red", false )
      val sgsOpenPort3 = SGOpenPortsDetail("warning", "eu-west-1", "name-3", "id-3", "vpc-3", "tcp", "8080", "Red", false )
      val sgsOpenPort4 = SGOpenPortsDetail("warning", "eu-west-1", "name-4", "id-4", "vpc-4", "tcp", "8080", "Red", false )
      val account = AwsAccount("id-1", "security", "security-role")
      EC2.updateSgsWithTags(account, List(sgsOpenPort1, sgsOpenPort2, sgsOpenPort3, sgsOpenPort4))((_, _) => Attempt.Right(result) ).value().toSet shouldBe
        Set(
        sgsOpenPort1.copy(stackId = Some("stack-id-1"), stackName = Some("stack-name-1")),
        sgsOpenPort2.copy(stackId = Some("stack-id-2"), stackName = Some("stack-name-2")),
        sgsOpenPort3.copy(stackId = Some("stack-id-3")),
        sgsOpenPort4)
    }
  }
```
It passes, but that does mean it is possible for the template to end up with a `Some('stackId)`, None` scenario, so we should probably use a getOrElse, or explicitly check on the existence of the stackName [on line 81 here](https://github.com/guardian/security-hq/pull/34/files#diff-7d6df40145447f3722e44ed02bf342c8R81)

Reply:
@katebee I put a mini code block that checks for stack name emptiness.